src/helpers/index.ts

```diff
@@ -28,3 +32,56 @@ export const shortAddress = (addr: string): string =>
   addr.length > 10 && addr.startsWith('0x')
     ? `${addr.substring(0, 6)}...${addr.substring(addr.length - 4)}`
     : addr
+
+export const sortDateAsc = (messA: { sent?: Date }, messB: { sent?: Date }) => {
+  if (!messA.sent) return 1
+  if (!messB.sent) return -1
+  return messA.sent.getTime() - messB.sent.getTime()
+}
+
+export const sortDateDesc = (
+  messA: { sent?: Date },
+  messB: { sent?: Date }
+) => {
+  if (!messA.sent) return 1
+  if (!messB.sent) return -1
+  return messB.sent.getTime() - messA.sent.getTime()
+}
+
+export const listenToStream = async <T>(
+  stream: Stream<T>,
+  callback: (value: T) => void
+) => {
+  for await (const value of stream) {
+    callback(value)
+  }
+}
+
+export const getLatestMessage = async (conversation: Conversation) => {
+  for await (const [message] of conversation.messagesPaginated({
+    pageSize: 1,
+    direction: SortDirection.SORT_DIRECTION_DESCENDING,
+  })) {
+    return message
+  }
+}
+
+export const createClient = async (signer?: Signer) => {
+  if (!signer) return
+  const storage = createInstance({
+    name: '@nft/chat',
+    storeName: 'xmtp-identities',
+    description: 'store identities for xmtp',
+  })
+
+  const address = await signer.getAddress()
+  const storageKey = address.toLowerCase()
+  if (!(await storage.getItem(storageKey))) {
+    const keys = await Client.getKeys(signer)
+    await storage.setItem(storageKey, keys)
+  }
+  const keys = await storage.getItem<Uint8Array>(storageKey)
+  return await (keys
+    ? Client.create(null, { privateKeyOverride: keys })
+    : Client.create(signer))
```

Comment:
No need to `return` and `await` here, can just return the promise directly:
```ts
return keys
  ? Client.create(null, { privateKeyOverride: keys })
  : Client.create(signer)
```
Usually a slightly performance benefit from doing so, explained here: https://eslint.org/docs/latest/rules/no-return-await

Reply:
One thing interesting with the await here is that the resolution of the promise is done in this function so if there is an error, the stack trace will show this exact function.
Not a big deal tho, will simplify the codebase

Reply:
[cd687e3](https://github.com/liteflow-labs/chat/pull/15/commits/cd687e3cfd294fca2e03a82f53658dcaaf22c072)